packages/flutter/lib/src/widgets/sensitive_content.dart

```diff
@@ -0,0 +1,311 @@
+
+import 'package:flutter/services.dart' show ContentSensitivity, SensitiveContentService;
+
+import 'async.dart' show AsyncSnapshot, ConnectionState, FutureBuilder;
+import 'container.dart';
+import 'framework.dart';
+
+/// Data structure used to track the [SensitiveContent] widgets in the
+/// widget tree.
+class ContentSensitivitySetting {
+  /// Creates a [ContentSensitivitySetting].
+  ContentSensitivitySetting();
+
+  /// The number of [SensitiveContent] widgets that have sensitivity level [ContentSensitivity.sensitive].
+  int _sensitiveWidgetCount = 0;
+
+  /// The number of [SensitiveContent] widgets that have sensitivity level [ContentSensitivity.autoSensitive].
+  int _autoSensitiveWidgetCount = 0;
+
+  /// The number of [SensitiveContent] widgets that have sensitivity level [ContentSensitivity.notSensitive].
+  int _notSensitiveWigetCount = 0;
+
+  /// Increases the count of [SensitiveContent] widgets with [sensitivityLevel] set.
+  void addWidgetWithContentSensitivity(ContentSensitivity sensitivityLevel) {
+    switch (sensitivityLevel) {
+      case ContentSensitivity.sensitive:
+        _sensitiveWidgetCount++;
+      case ContentSensitivity.autoSensitive:
+        _autoSensitiveWidgetCount++;
+      case ContentSensitivity.notSensitive:
+        _notSensitiveWigetCount++;
+      // ignore is safe because it protects this setting from tracking SensitiveContent
+      // widgets with an _unknown ContentSensitivity. _unknown is private to avoid
+      // developers using it as a SensitiveContent sensitivity level.
+      // ignore: no_default_cases
+      default:
+        throw FlutterError(
+          'Adding a SensitiveContent widget with ContentSensitivity $sensitivityLevel is unsupported by ContentSensitivitySetting',
+        );
+    }
+  }
+
+  /// Decreases the count of [SensitiveContent] widgets with [sensitivityLevel] set.
+  void removeWidgetWithContentSensitivity(ContentSensitivity sensitivityLevel) {
+    switch (sensitivityLevel) {
+      case ContentSensitivity.sensitive:
+        _sensitiveWidgetCount--;
+      case ContentSensitivity.autoSensitive:
+        _autoSensitiveWidgetCount--;
+      case ContentSensitivity.notSensitive:
+        _notSensitiveWigetCount--;
+      // ignore is safe because it protects this setting from tracking SensitiveContent
+      // widgets with an _unknown ContentSensitivity. _unknown is private to avoid
+      // developers using it as a SensitiveContent sensitivity level.
+      // ignore: no_default_cases
+      default:
+        throw FlutterError(
+          'Removing a SensitiveContent widget with ContentSensitivity $sensitivityLevel is unsupported by ContentSensitivitySetting',
+        );
+    }
+  }
+
+  /// Returns true if this class is currently tracking at least one [SensitiveContent] widget.
+  bool get hasWidgets =>
+      max(0, _sensitiveWidgetCount) +
+          max(0, _autoSensitiveWidgetCount) +
+          max(0, _notSensitiveWigetCount) >
+      0;
```

Comment:
Should you also assert that these values are not less than zero? Here and/or in removeWidgetWithContentSensitivity.